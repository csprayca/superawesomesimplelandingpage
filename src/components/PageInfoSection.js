import React, { Fragment } from 'react';

const PageInfoSection = ({
  title = 'Last minute laura',
  subTitle = 'Artist. Maker. Creator.'
}) => {
  return (
    <Fragment>
      <section>
        <h1>{title}</h1>
        <h2>{subTitle}</h2>
      </section>
      <style jsx>{`
        section {
          padding-left: 1em;
          display: flex;
          flex-direction: column;
          align-items: center;
          justify-content: center;
        }
        h1 {
          text-transform: uppercase;
        }
        h2 {
          font-weight: lighter;
        }
      `}</style>
    </Fragment>
  );
};

export default PageInfoSection;
