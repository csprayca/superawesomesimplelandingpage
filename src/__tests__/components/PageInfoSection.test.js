/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import PageInfoSection from '../../components/PageInfoSection';

describe('PageInfoSection component', () => {
  describe('given default settings', () => {
    it('works as expected', () => {
      const component = renderer.create(<PageInfoSection />);
      const renderTree = component.toJSON();
      expect(renderTree).toMatchSnapshot();
    });
  });

  describe('given proper input', () => {
    it('works as expected', () => {
      const component = renderer.create(
        <PageInfoSection title="Alex Kluew" subTitle="Make all the things!" />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
