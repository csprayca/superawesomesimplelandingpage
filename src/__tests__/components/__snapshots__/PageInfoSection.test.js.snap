// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageInfoSection component given default settings works as expected 1`] = `
<section
  className="jsx-3411386411"
>
  <h1
    className="jsx-3411386411"
  >
    Last minute laura
  </h1>
  <h2
    className="jsx-3411386411"
  >
    Artist. Maker. Creator.
  </h2>
</section>
`;

exports[`PageInfoSection component given proper input works as expected 1`] = `
<section
  className="jsx-3411386411"
>
  <h1
    className="jsx-3411386411"
  >
    Alex Kluew
  </h1>
  <h2
    className="jsx-3411386411"
  >
    Make all the things!
  </h2>
</section>
`;
